pkg/runner/command.go

```diff
@@ -77,8 +77,15 @@ func (rc *RunContext) setEnv(ctx context.Context, kvPairs map[string]string, arg
 	rc.Env[kvPairs["name"]] = arg
 }
 func (rc *RunContext) setOutput(ctx context.Context, kvPairs map[string]string, arg string) {
-	common.Logger(ctx).Infof("  \U00002699  ::set-output:: %s=%s", kvPairs["name"], arg)
-	rc.StepResults[rc.CurrentStep].Outputs[kvPairs["name"]] = arg
+	stepID := rc.CurrentStep
+	outputName := kvPairs["name"]
+	if outputMapping, ok := rc.OutputMappings[MappableOutput{StepID: stepID, OutputName: outputName}]; ok {
```

Comment:
If there is a mismatch/typo between an output mapping step ID and the actual composite step ID we get a exception/panic due to a null memory reference on line 88 below. eg.

```
outputs:
  tag_name:
    description: "tag to use for the docker image"
    value: ${{ steps.GET_BAG.outputs.tag_name }}

runs:
  using: "composite"
  steps:
    - name: Determine tag
      id: get_tag
      shell: bash
      run: echo "::set-output name=tag_name::latest" 
```

In my foolish case I mixed up '_' and '-' which was hard to spot!

However, it would be nice to have a more informative error than a panic.